app/models/concerns/benchmark_searching.rb

```diff
@@ -29,18 +29,26 @@ module BenchmarkSearching
       end
     }
 
-    scope :latest_supported, lambda {
-      SupportedSsg.latest_per_os_major.inject(none) do |supported, ssg|
-        supported.or(
-          where(ref_id: ssg.ref_id,
-                version: ssg.upstream_version || ssg.version)
-        )
-      end
+    scope :order_by_version, lambda {
+      order(
+        Arel.sql("string_to_array(benchmarks.version, '.')::int[] DESC")
+      )
     }
   end
 
   # class methods for benchmark searching
   module ClassMethods
+    def latest_supported
+      SupportedSsg.by_os_major.inject(none) do |union, (os_major_version, ssgs)|
+        ssg_versions = ssgs.map { |ssg| ssg.upstream_version || ssg.version }
+        union |
+          where(version: ssg_versions)
```

Comment:
I've pushed this as another commit to the PR. I'm going to merge it and submit a stable PR so QEs have as much time as possible to test before the deployment.

Reply:
Nice. Thanks.

Initially I wanted to do a `UNION` which isn't natively supported by Rails (only via some "hacks" with Arel).

Reply:
@akofink doesn't the `find_by` return a model immediately (instead of a query/Relation). If that's the case, using `where` and `limit` would keep it all within one query.

Reply:
Ah! That's a good point. I can submit a small follow up patch.

Reply:
Patched in https://github.com/RedHatInsights/compliance-backend/pull/847